framework/src/main/java/com/github/dbmdz/flusswerk/framework/exceptions/SkipProcessingException.java

```diff
@@ -0,0 +1,71 @@
+package com.github.dbmdz.flusswerk.framework.exceptions;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableList;
+
+import com.github.dbmdz.flusswerk.framework.model.Message;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/** Indicates that processing should be skipped from the point on where this exception is thrown. */
+public class SkipProcessingException extends RuntimeException {
+
+  private final List<Message> outgoingMessages = new ArrayList<>();
+
+  /**
+   * Skip processing with an explanation and a cause.
+   *
+   * @param message The explanation why skipping makes sense here.
+   * @param cause The cause why skipping should happen.
+   */
+  public SkipProcessingException(String message, Throwable cause) {
+    super(message, cause);
+  }
+
+  /**
+   * Skip processing with an explanation of why.
+   *
+   * @param message Why should processing be skipped?
+   */
+  public SkipProcessingException(String message) {
+    super(message);
+  }
+
+  /**
+   * Add a message to send to the next workflow job.
+   *
+   * @param message The message to send.
+   * @return <code>this</code> for a fluent interface.
+   */
+  public SkipProcessingException send(Message message) {
```

Comment:
Do we need to explicitly state the single-arg use case when we have a varargs overload of the same method?

Reply:
Good point. Likely not!